enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,314 @@
+
+* `logs.app` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
+* `logs.infra` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infra" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
+
+Future types may be:
+* `events.kubernetes` - Kubernetes events
+* `audit.container-engine` - Container engine audit events
+
+There are no assumptions regarding whether or not an endpoint is deployed on or off cluster.  Endpoints off-cluster may require adminstrators to perform additional actions in order for logs to be forwarded (e.g. secret creation, opening port, enable global proxy configuration) 
+Following is the list of supported endpoint types for this proposal:
+
+* Elasticsearch
+* Fluent secure-forward
+
+### User Stories
+
+#### As an OKD admin, I want to deploy only a collector, and have it ship logs off-cluster to my own aggregation solution
+
+This is a typical example of organizations that desires to re-use their existing enterprise log solution.  We will succeeded if we are able to write logs to their logging service.
+
+#### As an OKD admin, I want to aggregate application logs on-cluster and infra logs off-cluster
+
+This is an example of an OKD cluster hosting solution where several organizations are each provided with a dedicated cluster.  The organization requires access to application container logs but the host requires access to the infra structure logs.
+
+### Implementation Details
+
+#### Assumptions
+
+* A pipeline shall support at least one target endpoint. 
+* A pipeline may support multiple target endpoints.
+* Endpoints that require secrets which are not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
+* Secrets shall have keys of: cert,key,cacert which point to the respective certificates for which they represent
+
+#### Security
+* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: cert, key, cacert
+* Endpoints defined without a secret will use unsecure communication
+
+#### Log Message Routing
+* Undeclared source messages default to being routed to the internal logstore if you have one, vanishes to devnull otherwise
+* Declared source messages are routed to the defined service
+
+For example, If you have specified a source and target for `logs.app`, but did not specify anything for `logs.infra`, then either `logs.infra` logs will not be collected at all, or they will be discarded.
+
+
+### Risks and Mitigations
+
+## Design Details
+
+The `cluster-logging-operator` will use the forwarding API to generate a collector configuration and mount any secrets into the collector pod.
+
+### API Specification
+The proposed forwarding API:
```

Comment:
As previously requested:  please split the log sources into different CRs so that we can properly secure them in a managed environment.

We need to allow dedicated-admins to (optionally) configure logs.app, while at the same time preventing dedicated-admins from ever modifying logs.infra.

This falls into our "Compliance" bucket of requirements.

Reply:
> As previously requested: please split the log sources into different CRs so that we can properly secure them in a managed environment.
> 
> We need to allow dedicated-admins to (optionally) configure logs.app, while at the same time preventing dedicated-admins from ever modifying logs.infra.
> 
> This falls into our "Compliance" bucket of requirements.
We explicitly identified this proposal as "Tech preview" because @sichvoge has suggested the API may need to change, partially for exactly this reason.  I'm not certain we will come to a consensus in the initial release but we do intend to resolve for GA